source/credentials_provider_profile.c

```diff
@@ -428,11 +455,35 @@ struct aws_credentials_provider *aws_credentials_provider_new_profile(
         goto on_finished;
     }
     const struct aws_profile_property *role_arn_property = aws_profile_get_property(profile, s_role_arn_name);
+    bool profile_contains_access_key = aws_profile_get_property(profile, s_access_key_id_profile_var);
+    bool profile_contains_secret_access_key = aws_profile_get_property(profile, s_secret_access_key_profile_var);
+    bool profile_contains_credentials = profile_contains_access_key || profile_contains_secret_access_key;
```

Comment:
should this be end? cred should not be valid if it only contains one of the 2

Reply:
It's an `Or` condition so that we don't treat the profile as an STS profile, even if partial creds are present at https://github.com/awslabs/aws-c-auth/pull/224/files#diff-f7cca12a8538a30712ad4b18e39bb61a240457b820982e916f6f27e76628c64fR477. Later, we fail the creation as invalid if either of them is missing at https://github.com/awslabs/aws-c-auth/pull/224/files#diff-f7cca12a8538a30712ad4b18e39bb61a240457b820982e916f6f27e76628c64fR481. This handles cases where the config file is as follows:

```

[profile userA]
role_arn =  arn
source_profile = userB

[profile userB]
role_arn = arn
aws_access_key_id = accessJey
source_profile = userC

[profile userC]
aws_access_key_id = acess_key
secret_access_key_id = secret
```
We should fail the request at profileB due to partial credentials instead of using creds from profileC. I copied this behavior from the AWS CLI.